### ISSUE TITLE
Please make a version 91.13

### ISSUE THREAD

**Issue by greyrat116** [amurzeau/debian-autobuild#1]
These options don't work or slow with SSE and OpenGL 2.0, switch them by default

gfx.ycbcr.accurate-conversion - true
false for SSSE3 and AVX2 
https://bugzilla.mozilla.org/show_bug.cgi?id=1256475
https://bug1256475.bmoattachments.org/attachment.cgi?id=8758540

layers.async-pan-zoom.enabled - false
media.av1.enabled - false
media.cubeb.sandbox - false
media.resume-bkgnd-video-on-tabhover - false

media.rdd-process.enabled - false
https://wiki.mozilla.org/Security/Sandbox/Process_model
https://www.reddit.com/r/AV1/comments/b3bcft/firefox_nightly_now_uses_dav1d_021/

security.sandbox.content.level - 0 or 1 ( 2, 3 or 4 - no sound in Antix)
widget.non-native-theme.enabled - false
gfx.webrender.force-disabled - true
gfx.webrender.software.opengl - true
dom.ipc.processCount - 1
browser.tabs.remote.warmup.enabled - false
browser.tabs.remote.separatePrivilegedMozillaWebContentProcess - false
general.smoothScroll - false

**Comment by amurzeau** [comment]
I can't blindly change all these options, I don't firefox details enough to do this (and my goal is primarily to make firefox run on non-sse CPUs).

What's your machine ? Can you tell how much they improve performance ?

**Comment by greyrat116** [comment]
Sempron 2400+ 462 socket, 768mb RAM, Antix 21.

Run the original firefox-esr_91.13.0esr-1~deb11u1_i386.deb
The browser falls and does not open the pages, this is fix if you сreate «New Private Window» and switch widget.non-native-theme.enabled - false.
widget.non-native-theme.enabled - does not work with SSE2

media.cubeb.sandbox - does not work with SSE2 (crash view video)

security.sandbox.content.level - 0 or 1 ( 2, 3 or 4 - no sound Alsa in Antix)

media.av1.enabled - Youtube often offers this very heavy codec. Pentium 3 can't even play 144p.

gfx.webrender.force-disabled - the old Basic renderer is much faster than Webrender (Software). Hardware Webrender works with OpenGL 3.3+. Motherboards with SSE only have an AGP slot.
Switch this option to true and see how much faster the browser will work.

gfx.webrender.software.opengl - if the graphics card supports OpenGL2.0, webrender will be partially hardware-based "WebRender (Software OpenGL)" , but it will be slower than "Basic" or "OpenGL" acceleration. This option is for those who want to enable Webreder (gfx.webrender.force-disabled false) with an old graphics card.

layers.async-pan-zoom.enabled - if the processor has multiple cores and the video card is running with hardware acceleration, this option may improve smoothness scrol. But it consumes RAM significantly more and is slowly working on a single core processor. Switch this option to false and see how the browser will work.

dom.ipc.processCount - many processes use more RAM. Switch this option to 1 or 10 and see how the browser will work.

browser.tabs.remote.warmup.enabled , media.resume-bkgnd-video-on-tabhover - this functions is not for weak processors
https://www.askvg.com/tip-how-to-disable-tab-warming-feature-in-mozilla-firefox/
https://bugzilla.mozilla.org/show_bug.cgi?id=1274919

general.smoothScroll - smooth scrol good work well if hardware acceleration is turned on. SSE computers without such video cards.

browser.tabs.remote.separatePrivilegedMozillaWebContentProcess - additional useless process
https://bugzilla.mozilla.org/show_bug.cgi?id=1472212

**Comment by greyrat116** [comment]
widget.non-native-theme.enabled , media.cubeb.sandbox - does not work WITHOUT SSE2
sorry

**Comment by amurzeau** [comment]
Thanks for your detailed analysis.

Did you tested the build I've made ? Has it all the crashes you get with 91.13 from AntiX ?
With the default options, I don't get crashes on Athlon XP (SSE only, no SSE2) even when opening a video on youtube (which is too slow but still won't crash)

Maybe some of these changes are good to keep as they will improve performance, but the one that prevents crashes might not be all needed.

**Comment by greyrat116** [comment]
> Did you tested the build I've made ?

Yes 91.11. No sound in Antix 21 if security.sandbox.content.level 2, 3 or 4. 
0 or 1 fix the sound.
The rest works without bugs


> Has it all the crashes you get with 91.13 from AntiX ?

Yes original Debian Firefox ESR 78.0 - 91.13. With SSE2 processor these errors are not present in Antix.


> which is too slow

Many functions in the new Firefox really accelerate the work of the browser, but for their work they require a modern processor or a modern video card.
In the first message I wrote gfx.ycbcr.accurate-conversion 
true - optimization for MMX/SSE
false (default) - SSSE3 and AVX2 optimization

uBlock 1.17+ works with WASM, which works with SSE2
Install the latest version of Ublock 1.44 
and then the last version without WASM 1.16.20 https://addons.mozilla.org/firefox/downloads/file/1056733/ublock_origin-1.16.20-an+fx.xpi
Is the difference very noticeable? :)
On my eeePC900  Celeron M 900MHz with SSE2, uBlock 1.44 works as quickly as 1.16
**SSE2 software bypass is very slow for SSE. It is better to turn off such a code in unnecessary places**

widget.non-native-theme.enabled - The new design of the scrolling from Mozilla who needs SSE2. It is quite possible to get by system scrol which does not need SSE2

media.cubeb.sandbox and media.rdd.... - Sandbox for media codecs, they work slowly everywhere, and I turn them off not only on SSE

security.sandbox.content.level - The larger the number, the safer, but the slower it works.

**Comment by greyrat116** [comment]
By the way. 
In Firefox 102, you can not turn off the webrender and reduce the number of processes. Version 102 will work much slower on the old computer than configured Firefox 91
Therefore, I ask you to configure 91

**Comment by amurzeau** [comment]
I've no sound issues with the default option and with [firefox-esr/91.12.0esr-1+nosse2_deb11u1](https://github.com/amurzeau/debian-autobuild/releases/tag/firefox-esr%2F91.12.0esr-1%2Bnosse2_deb11u1).
I've tried with everything set as you said, and the memory usage seems better indeed, either it is more swap friendly, are much less consuming. Playing a 360p video on youtube is still slow but a bit faster than without options.

 browser.tabs.remote.separatePrivilegedMozillaWebContentProcess doesn't seem to work, I always have a 4 processes, maybe it is not working anymore since some version < 91.

I may add a build with these options preconfigured but later, as I'm busy on making a patch to have firefox build working without SSE on official Debian.

**Comment by greyrat116** [comment]
If your video card supports OpenGL 2.0 (Radeon 9500 or newer, GeForce 6200 or newer) you can enable hardware acceleration and the browser will work much faster.
gfx.webrender.force-disabled - true
layers.acceleration.force-enabled - true

> I may add a build with these options preconfigured but later, as I'm busy on making a patch to have firefox build working without SSE on official Debian.

Do it when it suits you.
